public/app/features/explore/RichHistory/RichHistoryContent.tsx

```diff
@@ -0,0 +1,244 @@
+      position: absolute;
+    `,
+    selectors: css`
+      display: flex;
+      justify-content: space-between;
+    `,
+    multiselect: css`
+      width: 60%;
+      .gf-form-select-box__multi-value {
+        background-color: ${bgColor};
+        padding: ${theme.spacing.xxs} ${theme.spacing.xs} ${theme.spacing.xxs} ${theme.spacing.sm};
+        border-radius: ${theme.border.radius.sm};
+      }
+    `,
+    sort: css`
+      width: 170px;
+    `,
+    sessionName: css`
+      display: flex;
+      align-items: flex-start;
+      justify-content: flex-start;
+      margin-top: ${theme.spacing.lg};
+      h4 {
+        margin: 0 10px 0 0;
+      }
+    `,
+    heading: css`
+      font-size: ${theme.typography.heading.h4};
+      margin: ${theme.spacing.md} ${theme.spacing.xxs} ${theme.spacing.sm} ${theme.spacing.xxs};
+    `,
+  };
+});
+
+export function RichHistoryContent(props: RichHistoryContentProps) {
+  const {
+    datasourceFilters,
+    onSelectDatasourceFilters,
+    queries,
+    onlyStarred,
+    onChangeSortOrder,
+    sortOrder,
+    onChangeRichHistoryProperty,
+    activeDatasourceOnly,
+    activeDatasourceInstance,
+    retentionPeriod,
+  } = props;
+
+  const [sliderRetentionFilter, setSliderRetentionFilter] = useState<number[]>([0, retentionPeriod]);
+
+  const theme = useTheme();
+  const styles = getStyles(theme, onlyStarred);
+  const exploreDatasources = getExploreDatasources()?.map(d => {
+    return { value: d.value!, label: d.value!, imgUrl: d.meta.info.logos.small };
+  });
+  const listOfDatasourceFilters = datasourceFilters?.map(d => d.value);
+
+  /* If user selects activeDatasourceOnly === true, set datasource filter to currently active datasource.
+   *  Filtering based on datasource won't be available. Otherwise set to null, as filtering will be
+   * available for user.
+   */
+  useEffect(() => {
+    activeDatasourceOnly && activeDatasourceInstance
+      ? onSelectDatasourceFilters([{ label: activeDatasourceInstance, value: activeDatasourceInstance }])
+      : onSelectDatasourceFilters(null);
+  }, [activeDatasourceInstance, activeDatasourceOnly]);
+
+  /* selectorsAndSorter id used several times, therefore extracted to variable */
+  const selectorsAndSorter = () => (
+    <div className={styles.selectors}>
+      {!activeDatasourceOnly && (
+        <div className={styles.multiselect}>
+          <Forms.Select
+            isMulti={true}
+            options={exploreDatasources}
+            value={datasourceFilters}
+            placeholder="Filter queries for specific datasources(s)"
+            onChange={onSelectDatasourceFilters}
+          />
+        </div>
+      )}
+      <div className={styles.sort}>
+        <Forms.Select
+          options={sortOrderOptions}
+          placeholder="Sort queries by"
+          onChange={e => onChangeSortOrder(e.value as SortOrder)}
+        />
+      </div>
+    </div>
+  );
+
+  /* If in Starred tab, render following content */
+  if (onlyStarred) {
```

Comment:
Feels a bit weird we have a component that basically renders 2 different things based on an if. Wouldn't it be clearer to just have 2 different components one for each tab (starred/history tab)?

Reply:
👍 I totally agree. The goal was to reuse as much code as possible, but it ended up as one huge component that is doing 2 separate things. I have split it into 2 components.